python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -357,6 +357,11 @@ def compute_child_asset_slice(
             ),
         )
 
+    def compute_materialized_asset_slice(self, asset_key: "AssetKey") -> "AssetSlice":
```

Comment:
Yeah good point..

`has_materialization_or_observation` is a terrible name which confuses everyone who crosses it, but when called on a non-source asset, it will only query for materializations (and conversely will only query for observations when called on a source asset).

That being said, if this is used in an `any_deps_match` context, then this condition could be applied against a source asset, which wouldn't make sense.

Ideally, there'd be some word that encompasses "materialized or observed", but maybe the easiest thing to do is just stick with the AMP-world naming scheme, where we just compute the inverse of this and call it "missing".

The definition there would be (essentially):

- an asset partition of a materializable asset is missing if it has never been materialized, and an asset partition of an observable asset is missing if it has never been observed

That does leave the question of what to do about non-materializable / non-observable assets, and I think the easiest thing would be to just exclude them from the dependencies that we search over in AnyDepsMatch.